### ISSUE TITLE
Maintainability going forward: one possible, easy solution

### ISSUE THREAD

**Issue by itamarst** [psf/cachecontrol#260]
Since you have very little time to maintain this, like we all do, one thing I've successfully done, inspired by https://lukasa.co.uk/2016/04/One_Of_The_Team/ via Hynek Schlawack: anyone who lands a PR gets asked if they want commit access.

In particular, same company that is getting me to fix the memory usage bug in `pip` needed some bug fixes in sparkmagic, which had huge pile-up of PRs and basically zero maintainers. So I finagled commit access, merged some, and then recruited those people to be maintainers, since I didn't actually know anything about Spark :) Basically just "hey do you want to be maintainer, you just need to agree to code of conduct, and try to PRs with reviews instead of pushing directly." You can enforce the latter in repo settings I think?

The result was pretty successful; it hasn't gotten a huge development push, but it is being actively maintained by people who are not me (https://github.com/jupyter-incubator/sparkmagic/commits/master?before=ac0852cbe88a41faa368cf1e1c89045a2de973bf+35&branch=master).

And maybe in this case you want to be a little more careful, given the potential for supply chain attack on `pip`, but you already have a bunch of commits/PRs from people who are credible and apparently motivated to contribute.

**Comment by davegaeddert** [comment]
Just curious if there have been any updates on the release process/timing? I know everyone is busy :) I've been trying to use my fork / a git repo source but, because of the way my code is packaged/distributed, I really need to get back to a release from PyPI... I can certainly look into alternatives for myself but was hoping a new release would be coming soon?! Thanks for your work on this!

**Comment by itamarst** [comment]
There have been some releases with some fixes, separate from this issue. Is there a specific fix you want to get from git that isn't released?

**Comment by davegaeddert** [comment]
Thanks for the quick response @itamarst. Basically everything since the last release, but specifically this PR  https://github.com/ionrock/cachecontrol/pull/269.

**Comment by itamarst** [comment]
There's an outstanding PR for first part of automation, and then when that's in I will do PR for the rest of release automation. But it's up to ionrock to merge, and he doesn't have a lot of time, so... I'm sure it'll happen eventually.

In short term, you can have pip install off of this git repo instead of your fork?

**Comment by davegaeddert** [comment]
Fair enough.

I'll take another look at the workarounds. I've got a bit of a unique situation at the moment where I'm using this as a dependency for something that is both 1) also a pypi package, and 2) being added to a private zip which is delivered to customers. I've got it as `https://github.com/ionrock/cachecontrol/archive/7815847.zip` right now but that can't be a used for a dependency on pypi (`HTTP Error 400: Invalid value for requires_dist. Error: Can't have direct dependency: 'CacheControl[filecache,redis] @ https://github.com/ionrock/cachecontrol/archive/7815847.zip'` [here](https://github.com/dropseed/pullapprove/runs/4475377012?check_suite_focus=true#step:3:158)), and when I had it as the git ref then I couldn't package my private zip (I think poetry export -> requirements.txt may have been the hangup there). I might just "vendor"/copy it in directly if I can't find another way to make it work for now...

Thanks for getting back to me — let me know if a release sneaks its way out 😉
